nx/lib/nx.ex

```diff
@@ -6658,6 +6658,217 @@ defmodule Nx do
     impl!(tensor).cholesky(out, tensor)
   end
 
+  @doc """
+  Calculates the p-norm of a tensor.
+
+  For the 0-norm, the norm is the number of non-zero elements in the tensor.
+
+  ### Examples
+
+    iex> Nx.norm(Nx.tensor([3, 4]))
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: 1)
+    #Nx.Tensor<
+      f64
+      7.0
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4]), ord: :inf)
+    #Nx.Tensor<
+      s64
+      4
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4]), ord: :neg_inf)
+    #Nx.Tensor<
+      s64
+      3
+    >
+
+    iex> Nx.norm(Nx.tensor([3, -4, 0, 0]), ord: 0)
+    #Nx.Tensor<
+      u64
+      2
+    >
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: :fro)
+    ** (RuntimeError) expected a 2-D tensor. Got a 1-D tensor.
+
+    iex> Nx.norm(Nx.tensor([3, 4]), ord: :nuc)
+    ** (RuntimeError) expected a 2-D tensor. Got a 1-D tensor.
+
+    iex> Nx.norm(Nx.tensor([[3, -1], [2, -4]]), ord: -1)
+    #Nx.Tensor<
+      s64
+      5
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: 1)
+    #Nx.Tensor<
+      s64
+      6
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: :neg_inf)
+    #Nx.Tensor<
+      s64
+      5
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, -2], [2, -4]]), ord: :inf)
+    #Nx.Tensor<
+      s64
+      6
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 0], [0, -4]]), ord: :fro)
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 0], [0, -4]]))
+    #Nx.Tensor<
+      f64
+      5.0
+    >
+
+    iex> Nx.norm(Nx.tensor([[3, 4], [0, -4]]), axes: [1], keep_axes: true)
+    #Nx.Tensor<
+      f64[2][1]
+      [
+        [5.0],
+        [4.0]
+      ]
+    >
+
+  ### Caveats
+
+  For big values of p, f64 rounding errors come into play
```

Comment:
Could we use a numerically stable implementation? https://timvieira.github.io/blog/post/2014/11/10/numerically-stable-p-norms/

Reply:
That's a great suggestion! I'll try make the proper adjustments for this.
Do we have a function that yields the maximum value for each datatype?
If not, maybe we should leave this as a technical debt to avoid hardcoding any values. I'm thinking of something like `Nx.max_value(1.0)` yielding `1.0e300`

Reply:
Yes, there is `Nx.Type.min_value_binary` which you can put into a tensor by calling `Nx.from_binary`.